packages/flutter/lib/src/material/flexible_space_bar.dart

```diff
@@ -97,6 +107,12 @@ class FlexibleSpaceBar extends StatefulWidget {
   /// Typically an [Image] widget with [Image.fit] set to [BoxFit.cover].
   final Widget? background;
 
+  /// Shown behind the [title] when expanded.
+  ///
+  /// Compared to `background`, can build background widget more flexibly.
+  final Widget Function(BuildContext context, FlexibleSpaceBarSettings settings,
+      BoxConstraints constraints)? backgroundBuilder;
```

Comment:
This could be confusing, do you think folks should migrate to only use backgroundBuilder? Or should both background and backgroundBuilder exist together? When should one be used over the other?

Reply:
`background` and `backgroundBuilder` cannot be used at the same time, the user can choose to use either one, it seems that we should modify the documentation to make this property better understood by others.